client/src/types.ts

```diff
@@ -91,3 +91,21 @@ export interface Feedback {
   rating: number;
   ratingError?: string;
 }
+
+export type Validator = {
+  required: (value: string) => boolean;
+  requiredRating: (value: string) => boolean;
+  string: (value: string) => boolean;
+  minLength: (value: string, param: number) => boolean;
+  maxLength: (value: string, param: number) => boolean;
+  validateEmail: (value: string) => boolean;
+};
+
+export type Errors = {
+  required: (value: string) => string;
+  requiredRating: (value: string) => string;
+  string: (value: string) => string;
+  minLength: (value: string, param: string) => string;
+  maxLength: (value: string, param: string) => string;
+  validateEmail: (value: string) => string;
+};
```

Comment:
both types can be defined more elegant way like this:
export type Validator = Record<Validations, (value: string, param?: number) => boolean>